musica/artista/tests.py

```diff
@@ -0,0 +1,207 @@
+        self.assertEqual(num_registros, len(respuesta.data))
+
+    def test_comparar_info_lista_vs_info_BD(self):
+        """
+        Comprobar que la información que se despliega es igual a la contenida en la BD
+        """
+        respuesta_get = self.client.get(self.url_respuesta)
+
+        for x in respuesta_get.data:
+            artista = Artista.objects.get(id=x["id"])
+            self.assertEqual(x["nombre"], artista.nombre)
+            self.assertEqual(x["nacionalidad"], artista.nacionalidad)
+            self.assertEqual(x["genero"], artista.genero)
+
+            for i in x["disquera"]:
+                disquera = Disquera.objects.get(id=i["id"])
+                j = artista.disquera.get(id=disquera.pk)
+                self.assertEqual(j.id, disquera.pk)
+
+    def test_confirmar_nuevo_registro_en_BD(self):
+        """
+        Comprobar que el numero de elementos despues del insert sea igual al numero de elementos inicial
+        en la BD + 1
+        """
+        self.client.login(username="testuser1", password="1X<ISRUkw+tuK")
+
+        data_info = {
+            "usuario": self.test_user1.pk,
+            "nombre": "Nuevo artista",
+            "nacionalidad": "Español",
+            "fecha_nacimiento": "2005-05-15",
+            "disquera_entrada": [self.disquera.id],
+            "genero": "ROCK",
+        }
+
+        num_elementos_pre_post = Artista.objects.count()
+        respuesta = self.client.post(self.url_respuesta, data_info)
+        num_elementos_post_post = Artista.objects.count()
+        self.assertEqual(num_elementos_pre_post + 1, num_elementos_post_post)
+        self.assertEqual(201, respuesta.status_code)
+
+    def test_comparar_info_registrada_vs_info_en_BD(self):
+        """
+        Comprobar que la información que se envió a guardar sea igual a la almacenada en la BD
+        """
+        self.client.login(username="testuser1", password="1X<ISRUkw+tuK")
+
+        data_info = {
+            "usuario": self.test_user1.pk,
+            "nombre": "Nuevo artista",
+            "nacionalidad": "Español",
+            "fecha_nacimiento": datetime.date(2010, 1, 1),
+            "disquera_entrada": [self.disquera.id],
+            "genero": "ROCK",
+        }
+
+        respuesta_post = self.client.post(self.url_respuesta, data_info)
+        self.assertEqual(201, respuesta_post.status_code)
+        respuesta_consulta = Artista.objects.get(id=respuesta_post.data["id"])
+        self.assertEqual(data_info["nombre"], respuesta_consulta.nombre)
+        self.assertEqual(data_info["nacionalidad"], respuesta_consulta.nacionalidad)
+        self.assertEqual(data_info["fecha_nacimiento"],
+                         respuesta_consulta.fecha_nacimiento)
+        self.assertEqual(data_info["genero"], respuesta_consulta.genero)
+
+        for i in respuesta_post.data["disquera"]:
+            disquera = Disquera.objects.get(id=i["id"])
+            for j in data_info["disquera_entrada"]:
```

Comment:
¿Qué pasa si agregas más de un elemento a tu lista de disquera_entrada?

Reply:
La prueba falla porque estamos recorriendo primero la respuesta y luego la información que ingresamos, entonces, estamos comparando el primer valor del primer "for* contra los "n" valores que tenga el segundo "for", por lo que la prueba en la segunda iteración en el "for" anidado falla.